packages/core/src/fhir-deduplication/resources/allergy-intolerance.ts

```diff
@@ -146,14 +176,44 @@ function isKnownAllergy(coding: Coding, text?: string | undefined) {
   return isValid;
 }
 
+function isUnknownAllergy(substance: CodeableConcept | undefined): boolean {
+  if (!substance) return false;
+
+  const { text, coding } = substance;
+
+  if (text && isUnknownAllergyText(text)) {
+    return true;
+  }
+
+  if (coding?.some(coding => isUnknownAllergyDisplay(coding.display))) {
+    return true;
+  }
+
+  return false;
+}
+
 function isUnknownAllergyText(text: string | undefined) {
   return text && blacklistedSubstanceDisplays.includes(text.toLowerCase().trim());
 }
 
+function isUnknownAllergyDisplay(display: string | undefined): boolean {
```

Comment:
why is this needed? the function above does the same thing if you pass text or display